xgcm/grid_ufunc.py

```diff
@@ -510,6 +522,9 @@ def apply_as_grid_ufunc(
     keep_coords: bool = True,
     dask: Literal["forbidden", "parallelized", "allowed"] = "forbidden",
     map_overlap: bool = False,
+    other_component: Union[
+        None, Dict[str, xr.DataArray], Sequence[Dict[str, xr.DataArray]]
```

Comment:
Is this what you were referring to with lists of dicts?

Reply:
Yes exactly